cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/hdfs/HdfsDelegationTokenRefresherConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.linkedin.openhouse.cluster.storage.hdfs;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+/**
+ * Configuration class to conditionally enable/disable creation of {@link
+ * HdfsDelegationTokenRefresher} bean and enable delegation token refresh scheduling.
+ */
+@Slf4j
+@Configuration
+@EnableScheduling
+public class HdfsDelegationTokenRefresherConfig {
+
+  @Autowired private HdfsStorage hdfsStorage;
+
+  private static final String HDFS_TOKEN_REFRESH_ENABLED = "token.refresh.enabled";
+
+  /**
+   * Conditionally provide the HdfsDelegationTokenRefresher bean if the parameter for token refresh
+   * is enabled in the HdfsStorage properties. The relevant configuration in the cluster YAML file
+   * is as follows:
+   *
+   * <pre>
+   * cluster:
+   *   storages:
+   *     hdfs:
+   *       parameter:
+   *         token.refresh.enabled: true
+   * </pre>
+   *
+   * @return HdfsDelegationTokenRefresher
+   */
+  @Bean
+  public HdfsDelegationTokenRefresher getDelegationTokenRefresher() {
+    if (!hdfsStorage.isConfigured()) {
+      log.debug(
+          "Hdfs storage is not configured, ignoring HdfsDelegationTokenRefresher bean creation");
+      return null;
+    }
+    String refreshEnabled =
```

Comment:
Is there a need of this param if there's something as `hdfsStorage.isConfigured()` ?
I mean, not configured == not enabled. Is that sufficient?

Reply:
yes it is sufficient for `not configured == not enabled` case, but we need a yaml flag to turn the feature on/off if hdfs is configured